docs/configuration/config-file-reference.md

```diff
@@ -2679,6 +2679,12 @@ The `limits_config` configures default and per-tenant limits imposed by Cortex s
 # CLI flag: -querier.max-fetched-chunk-bytes-per-query
 [max_fetched_chunk_bytes_per_query: <int> | default = 0]
 
+# The maximum size of all labels in bytes that a query can fetch from each
+# ingester and storage. This limit is enforced in the querier and ruler only
+# when running Cortex with blocks storage. 0 to disable.
+# CLI flag: -querier.max-fetched-label-bytes-per-query
+[max_fetched_label_bytes_per_query: <int> | default = 0]
```

Comment:
I am wondering if it would be better if we deprecate `max_fetched_chunk_bytes_per_query` and have a new configuration that is simply called `max_fetched_data_bytes_per_query`, which limits the combination of labels and chunks. 

Reason for this suggestion is so that we don't make users think about what is "label" v.s. what is "chunk"; I this is simpler for users. 

What do you think @harry671003 @alanprot @friedrichg ?

Reply:
I was talking to @alanprot who wasn't convinced we really need this limit. But from the stress tests, I saw that queriers were OOM killed when fetching 2 million series with 10KB labels size. So we still need a way to control the total bytes fetched during a query.
It makes sense to combine them into a single limit.

It might also be possible to not enable other limits like, max_fetched_chunks, max_fetched_series etc if we enable the max_fetched_bytes_per_query limit.